include/IDriver.h

```diff
@@ -138,10 +138,17 @@ namespace video
                 const asset::SBlendParams& _blendParams,
                 const asset::SPrimitiveAssemblyParams& _primAsmParams,
                 const asset::SRasterizationParams& _rasterParams
-            ) {
+            )
+            {
                 return nullptr;
             }
 
+            //! Must be called by thread before last reference of `_pipeline` object is dropped by the thread.
```

Comment:
explain?

Reply:
read this commit's diff https://github.com/buildaworldnet/IrrlichtBAW/pull/358/commits/f0f24c05d32254c555cab669ce77e50fca8531bf
pipeline hash has to be removed from context's cache

Reply:
ok but this necessitates that I remove the `IGPURenderpassIndependentPipeline` with the same thread that created the cache entry.

This is unlike the old VAO where I could create `IGPUMeshDataFormatDesc` and remove it on any thread, the hard limit of total VAO could (plus a global clear function) would take care of removal.

Reply:
ah ok I got the English syntax now XD the point of removing calling this `remove` function is that pipeline's shaders GL names might be reused once shaders gets destroyed. So there is a possibility that some new pipeline would get the same hash while the old pipeline's hash still exist in the cache. This would end up in finding pipeline GL name consisting of not-anymore existing shaders (invalid GL names)

Reply:
> This is one I don't understand at all. Why are you talking about mesh data format desc and what vao has to do with all this?

Because my old API didn't have this threading requirement and thats how I want the new one to be.

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r336351500 do you get it now?

Reply:
> the point of removing calling this remove function is that pipeline's shaders GL names might be reused once shaders gets destroyed.

But if you're using smart_refctd_ptr in the per-context pipeline cache the old shaders (and their GL names) will not be destroyed until you remove the pipeline (and drop the shaders) hence you won't be able to get the same GL handles as long as the pipeline is in the cache.

Reply:
going that way, hash for pipeline can be just `smart_refctd_ptr<COpenGLRenderpassIndependentPipeline>`

Reply:
> going that way, hash for pipeline can be just

That will lead to pointless copious amounts of PRogram Pipeline and VAO objects under OpenGL ?